language/stdlib/modules/LibraAccount.move

```diff
@@ -186,40 +193,18 @@ module LibraAccount {
     public fun initialize(
         lr_account: &signer,
         dummy_auth_key_prefix: vector<u8>,
-    ) {
+    ) acquires AccountOperationsCapability {
         LibraTimestamp::assert_genesis();
         // Operational constraint, not a privilege constraint.
         CoreAddresses::assert_libra_root(lr_account);
+
         create_libra_root_account(
             copy dummy_auth_key_prefix,
         );
-        // Create the treasury compliance account
         create_treasury_compliance_account(
             lr_account,
             copy dummy_auth_key_prefix,
         );
-
-        assert(
-            !exists<AccountOperationsCapability>(CoreAddresses::LIBRA_ROOT_ADDRESS()),
-            Errors::already_published(EACCOUNT_OPERATIONS_CAPABILITY)
-        );
-        move_to(
-            lr_account,
-            AccountOperationsCapability {
-                limits_cap: AccountLimits::grant_mutation_capability(lr_account),
-            }
-        );
-
-        assert(
-            !exists<LibraWriteSetManager>(CoreAddresses::LIBRA_ROOT_ADDRESS()),
-            Errors::already_published(EWRITESET_MANAGER)
-        );
-        move_to(
-            lr_account,
-            LibraWriteSetManager {
-                upgrade_events: Event::new_event_handle<Self::UpgradeEvent>(lr_account),
-            }
-        );
```

Comment:
Unfortunately, this won't wok because emitting the event (which happens in `create_libra_root_account`) requires `<THIS_CODE>` to be run first, specifically, the `AccountOperationsCapability` part.

I actually felt good about moving this code into `create_libra_root_account` because the pattern we use elsewhere is to publish X role-specific resources in `create_X_account`. But certainly open to alternatives!

Reply:
ahhh I see! For some reason I didn't see that the event emission is happening in `make_account`. 

I'm fine with this as-is since I can't think of any other alternatives that wouldn't make the code more complicated.